File: Domain/Models/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Models
{
    //حجوزات
    public class Reservation
    {
        public Reservation()
        {
            Id = Guid.NewGuid();
        }
        public Guid Id { get; set; }
        public bool IsDeleted { get; set; } = false;
        public Guid UserId { get; set; }
        public User User { get; set; }
        public Guid ServiceId { get; set; }
        public Service Service { get; set; }
        public Guid AppointmentId { get; set; }
        public Appointment Appointment { get; set; }

    }
}

File: ScheduledServicesCenterUI/src/models/reservation.ts
import { Guid } from "guid-typescript";
import { User } from "./user";
import { Service } from "./service";
import { Appointment } from "./appointment";

export class Reservation {
    Id: Guid | undefined;
    IsDelete: boolean | undefined;
    UserId: Guid | undefined;
    User: User | undefined;
    ServiceId: Guid | undefined;
    Service: Service | undefined;
    AppointmentId: Guid | undefined;
    Appointment: Appointment | undefined;
}